blocks.go

```diff
@@ -2,47 +2,43 @@ package main
 
 import (
 	"context"
-	"fmt"
 	"notes-service/models"
 	notespb "notes-service/protorepo/noted/notes/v1"
+	"notes-service/validators"
 	"strconv"
 
-	"github.com/google/uuid"
 	"go.uber.org/zap"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	"google.golang.org/protobuf/types/known/emptypb"
 )
 
-type blocksService struct {
-	notespb.UnimplementedNotesAPIServer
-
-	logger *zap.Logger
-	repo   models.BlocksRepository
-}
-
-var _ notespb.NotesAPIServer = &notesService{}
-
-func (srv *blocksService) InsertBlock(ctx context.Context, in *notespb.InsertBlockRequest) (*notespb.InsertBlockResponse, error) {
-	_, err := uuid.Parse(strconv.Itoa(int(in.NoteId)))
+func (srv *notesService) InsertBlock(ctx context.Context, in *notespb.InsertBlockRequest) (*notespb.InsertBlockResponse, error) {
+	token, err := srv.authenticate(ctx)
 	if err != nil {
-		srv.logger.Error("invalid uuid :", zap.Error(err))
-		return nil, status.Errorf(codes.InvalidArgument, "could not insert block")
+		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
-
-	if in.Block.Data == nil || in.Index < 1 || in.Block.Type < 1 {
-		srv.logger.Error("invalid arguments")
-		return nil, status.Errorf(codes.InvalidArgument, "could not insert block")
+	err = validators.ValidateInsertBlockRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	//check si la note appartient a celui qui veut la modifier
+	note, err := srv.repoNote.Get(ctx, strconv.Itoa(int(in.NoteId)))
+	if err != nil {
+		srv.logger.Error("Note not found in database", zap.Error(err))
```

Comment:
No need to log here for two reasons, 1) the error returned by your endpoint will be logged by the gRPC logger. 2) It could be a different error than a not found error, if it's the case, the repository would have already logged it.

Reply:
The protobuf has a problem, we take a uint32 for note_id instead of a string, I've open a PR in protorepo to fix it in notes-service and protorepo : https://github.com/noted-eip/protorepo/pull/48